src/context-providers/builder/hook.ts

```diff
@@ -24,40 +23,37 @@ export const useBuilder = () => {
         [state.orderedIdentifiers]
     );
 
-    const addElement = useCallback(
-        (type: EElementType, setFocus?: boolean) => {
-            const existingIdentifiers = state.orderedIdentifiers.map(
-                (elementId) => elementId.internalId
-            );
-            const newElement: TElement = ElementObjectGenerator.generate(
-                type,
-                existingIdentifiers
-            );
-            const newOrderedIdentifiers = [
-                ...state.orderedIdentifiers,
-                { internalId: newElement.internalId },
-            ];
+    const addElement = useCallback((type: EElementType, setFocus?: boolean) => {
+        const existingIdentifiers = state.orderedIdentifiers.map(
+            (elementId) => elementId.internalId
+        );
+        const newElement: TElement = ElementObjectGenerator.generate(
+            type,
+            existingIdentifiers
+        );
+        const newOrderedIdentifiers = [
+            ...state.orderedIdentifiers,
+            { internalId: newElement.internalId },
+        ];
+
+        dispatch({
+            type: "add-element",
+            payload: {
+                element: newElement,
+                orderedIdentifiers: newOrderedIdentifiers,
+            },
+        });
 
+        if (setFocus) {
             dispatch({
-                type: "add-element",
+                type: "focus-element",
                 payload: {
                     element: newElement,
-                    orderedIdentifiers: newOrderedIdentifiers,
+                    isDirty: true,
                 },
             });
-
-            if (setFocus) {
-                dispatch({
-                    type: "focus-element",
-                    payload: {
-                        element: newElement,
-                        isDirty: true,
-                    },
-                });
-            }
-        },
-        [state.orderedIdentifiers]
-    );
+        }
+    }, []);
```

Comment:
with `useCallback`, the entire function will be memo-ised aka cached. this means that variables used within it will not be updated, hence if there are external non-argument variables used within it, they need to be added as dependencies. in this case there is a need to add `state.orderedIdentifiers`